src/main/java/org/openpnp/machine/reference/vision/ReferenceBottomVision.java

```diff
@@ -99,36 +98,52 @@ private static PartAlignmentOffset findOffsetsPreRotate(Part part, BoardLocation
-            camera=(Camera)pipeline.getProperty("camera");
-
-            angle = angleNorm(angleNorm(angle)
-                    + angleNorm((rect.size.width < rect.size.height) ? 90 + rect.angle : rect.angle));
-            // error is -angle
-            // See https://github.com/openpnp/openpnp/pull/590 for explanations of the magic
-            // values below. Reproduced here just in case:
-            
-            // For a 200 count stepper configured at 16 microsteps, a microstep is 0.1125 degrees.
-            // 360 / 200 / 16 = 0.1125
-            // 0.1125 / 2 = 0.0567 = half a microstep
-            // 0.0765 is chosen so that when adding to half a microstep it is just a bit more than a microstep.
-            // 0.0765 + 0.0567 = 0.1332
-
-            // The check is intended to not introduce further error if the error is small. 
-            // I believe it is to avoid rounding of microsteps, but I'm not clear on that.
-            
-            if (Math.abs(angle) > 0.0765) {
-                angle += 0.0567 * Math.signum(angle);
-            } // rounding
-
-            nozzle.moveTo(new Location(LengthUnit.Millimeters, Double.NaN, Double.NaN, Double.NaN,
-                    placementAngle + angle));
-
-            rect = processPipelineAndGetResult(pipeline, camera, part, nozzle);
-
-            Logger.debug("Result rect {}", rect);
-            Location offsets = VisionUtils.getPixelCenterOffsets(camera, rect.center.x, rect.center.y)
-                                          .derive(null, null, null, Double.NaN);
+        	Location offsets = new Location(location.getUnits());
+        	// Try getting a good fix on the part in multiple passes.
+        	for(int pass = 0;;) {
+	            RotatedRect rect = processPipelineAndGetResult(pipeline, camera, part, nozzle);
+	            camera=(Camera)pipeline.getProperty("camera");
+	            
+	            Logger.debug("Result rect {}", rect);
+	            
+	            // Create the offsets object. This is the physical distance from
+	            // the center of the camera to the located part.
+	            offsets = VisionUtils.getPixelCenterOffsets(camera, rect.center.x, rect.center.y);
+	            
+	            // OpenCV can only tell us the angle of the recognized rectangle in a   
+	            // range of 0° .. 90° as it has no notion of which rectangle side 
+	            // is the true "bottom" side for instance. The angle will abruptly wrap 
+	            // around from 90° to 0° (and vice versa), as another of the four sides 
+	            // becomes the "bottom" side, as perceived by OpenCV.    
+	            // We can assume that the part is never picked more than +/-45º rotated.
+	            // So we change the wrapping-around into one that produces a range from 
+	            // -45° .. +45°. See angleNorm():
+	            angle = angleNorm(location.getRotation() + rect.angle);
```

Comment:
Simplification of the angle offset calculation. `angleNorm()` is in effect a "phase shifted" modulo operation and if I'm not completely mistaken [Distributive Equivalence](https://en.wikipedia.org/wiki/Modulo_operation#Equivalences) applies:

> (a + b) mod n = [(a mod n) + (b mod n)] mod n.

Also the 
> `(rect.size.width < rect.size.height) ? 90 + rect.angle : rect.angle`

expression is pointless as it will be removed by `angleNorm()` being a mod 90° function.